models/public/mozilla-deepspeech-0.8.2/mds_convert_utils/parse_trie_v6.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import re
+import itertools
+
+from collections import OrderedDict
+
+from .struct_utils import parse_format
+from .parse_openfst import parse_openfst
+
+
+DEFAULT_ALPHABET = [b' abcdefghijklmnopqrstuvwxyz\''[i:i+1] for i in range(28)]
+
+
+def find_trie_v6_offset(data):
+    template = re.compile(b'EIRT.\0\0\0..{8}.{8}\xD6\xFD\xB2\x7E')
+    matches = template.finditer(data)
+    matches = list(itertools.islice(matches, 2))
+    if len(matches) == 0:
+        raise ValueError("Cannot find OpenFst TRIE dictionary")
+    if len(matches) > 1:
+        raise ValueError("Found more than one OpenFst TRIE signature in LM file, cannot decide which one is right")
+    return matches[0].start()
+
+
+def scorer_cut_trie_v6(data, trie_offset=None):
+    """
+    Cut scorer file from Mozilla DeepSpeech v0.7.x/v0.8.x into kenlm and trie sections.
+    """
+    if trie_offset is None:
+        trie_offset = find_trie_v6_offset(data)
+    kenlm_data, trie_data = data[:trie_offset], data[trie_offset:]
+    return kenlm_data, trie_data, trie_offset
+
+
+def trie_v6_extract_vocabulary(data, alphabet=None, base_offset=0, max_num_words=10000000):
+    if alphabet is None:
+        alphabet = DEFAULT_ALPHABET
+    fst, _ = parse_trie_v6(data, pos=0, base_offset=base_offset)
+    vocabulary = traverse_fst(fst, alphabet, max_num_words=max_num_words)
+    return vocabulary, fst['meta']
+
+
+def parse_trie_v6(data, pos=0, base_offset=0):
+    (magic, version, is_utf8_mode, alpha, beta), pos = parse_format('<Ii?dd', data, pos)
+    if magic != 0x54524945:  # b'EIRT' = reversed 'TRIE'
```

Comment:
Instead of having the interpretation in the comment, just parse the magic number as `4s` instead of `I` and then you can directly compare it to a bytestring.

Reply:
1) Removed the comment here. 2) And converted the magic number in `parse_opendst.py` into hexadecimal form as well.

Reply:
I don't understand, why not load the magic number as a bytestring, like I said?

Reply:
OK, done.

I didn't want to read an uint32 field as a string, this seems weird to me.